docs/user/pings/testing-custom-pings.md

```diff
@@ -0,0 +1,121 @@
+# Unit testing Glean custom pings
+
+Applications defining [custom pings](custom.md) can use use the strategy defined in this document to test these pings in unit tests.
+
+## General testing strategy
+
+The schedule of custom pings depends on the specific application implementation, since it is up to the SDK user to define the ping semantics. This makes writing unit tests for custom pings a bit more involved.
+
+One possible strategy could be to wrap the Glean SDK API call to send the ping in a function that can be mocked in the unit test. This would allow for checking the status and the values of the metrics contained in the ping at the time in which the application would have sent it.
+
+## Example testing of a custom ping
+
+Let us start by defining a custom ping with a sample metric in it. Here is the `pings.yaml` file:
+
+```yaml
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+$schema: moz://mozilla.org/schemas/glean/pings/1-0-0
+
+my_custom_ping:
+  description: >
+    This ping is intended to showcase the recommended testing strategy for
+    custom pings.
+  include_client_id: false
+  bugs:
+    - 1556985
+  data_reviews:
+    - https://bugzilla.mozilla.org/show_bug.cgi?id=1556985
+  notification_emails:
+    - custom-ping-owner@example.com
+
+```
+
+And here is the `metrics.yaml`
+
+```yaml
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+
+my_custom_ping:
+  sample_string:
+    type: string
+    lifetime: ping
+    description: >
+      An hashed and salted version of the Google Advertising ID from the device.
+      This will never be sent in a ping that also contains the client_id.
+    send_in_pings:
+      - my_custom_ping
```

Comment:
Sphinx has a way to [highlight specific lines in a source code listing](http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block).  Wish we could do that here, because it's really just this line that's important, but I don't think mdbook supports that.

Reply:
Ah, that would be neat. The only similar thing I found was [hiding code lines](https://github.com/rust-lang-nursery/mdBook/blob/master/book-example/src/format/theme/syntax-highlighting.md), but it's only available for rust code blocks.